pkg/tuple/tuple.go

```diff
@@ -198,6 +198,43 @@ func StringToUserProto(userKey UserString) *openfgav1.User {
 	}}}
 }
 
+// FromObjectOrUsersetProto returns a string from a ObjectOrUserset proto. Ex: 'user:maria' or 'group:fga#member'. It is
+// the opposite from StringToObjectOrUserset function.
+func FromObjectOrUsersetProto(obj *openfgav1.ObjectOrUserset) UserString {
+	switch obj.GetUser().(type) {
+	case *openfgav1.ObjectOrUserset_Userset:
+		us := obj.GetUser().(*openfgav1.ObjectOrUserset_Userset)
+		return fmt.Sprintf("%s:%s#%s", us.Userset.GetType(), us.Userset.GetId(), us.Userset.GetRelation())
+	case *openfgav1.ObjectOrUserset_Object:
+		us := obj.GetUser().(*openfgav1.ObjectOrUserset_Object)
+		return fmt.Sprintf("%s:%s", us.Object.GetType(), us.Object.GetId())
+	default:
+		panic("unsupported type")
```

Comment:
```suggestion
		return ""
```

Reply:
Know we've been trying to be mindful about panicking but I prefer error'ing early rather than later in the process if we return empty string.

Reply:
I think we need to revisit these in a follow up PR anyways. We're introducing yet another series of functions to go to/from strings and structs.

I think we should overload `UserString` with methods such as `IsValid` which does string validation checks and other methods like `ToUserProto` etc.. This would make better use of that type and consolidate code that is now being duplicated more and more.